### PR TITLE
[Feature] AFRICA

### PR DESCRIPTION
**Pull request by kilemensi**
## Description

This PR adds ability to

  - [x] Fetch documents datasets information from openAFRICA (and any other CKAN based instance) based on url received from the CMS. CKAN's [Action API](https://docs.ckan.org/en/2.9/api/) is used.
  - [x] Fetch documents  information from sourceAFRICA (and any other DocumentCloud based instance) based on url received from the CMS.
  - [x] Paginate and order documents and datasets.

Closes #102  

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

localhost_3000_data_datasets

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by KobbyMmo — COMMENTED** [review]
I realized the dataset isn't loading for this group link.

Screenshot 2021-09-22 at 8 38 24 AM
Screenshot 2021-09-22 at 8 40 28 AM

**Comment by kilemensi** [comment]
> I realized the dataset isn't loading for this group link.
> 

@KobbyMmo and the reason being?

File: src/pages/data/[[...slug]].js [hunk 1]
```diff
@@ -62,6 +64,24 @@ export async function getStaticProps({ params, preview, previewData }) {
...
+            items = await Promise.all(items.map(fetchSourceAfricaDocuments));
+          }
+          // A single url can contain multiple datasets/documents & hence the
+          // need for flat(1)
+          items = items.flat(1);
+
+          return { ...others, type, items };
+        }
+      )
+    )) || null;
```
Comment (KobbyMmo) [inline src/pages/data/[[...slug]].js:84]: @kilemensi doesn't this belong to `formatBlocksForSections` more?

**Review by KobbyMmo — APPROVED** [review]
Looks good to me @kilemensi  🎸 🎻 🎺 👯

**Comment by KobbyMmo** [comment]
> @KobbyMmo and the reason being?

The link in the cms was wrong

File: src/pages/data/[[...slug]].js (hunk 1 above)
Comment (kilemensi) [reply src/pages/data/[[...slug]].js:84]: Probably @KobbyMmo ... `formatBlocksForSections` just scares me!

Comment (KobbyMmo) [reply src/pages/data/[[...slug]].js:84]: aww, the old man is afraid of bytes. 😂😂

File: src/utils/fetchOpenAfricaDatasets.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      if (apiPath) {
+        const json = await fetchJson(`${url.origin}${apiPath}`);
+        // group datasets are in result.results while individual dataset
+        // is in result
+        const results = json.result.results || [json.result];
+        return extractDatasets(url.origin, results);
+      }
+    }
+  } catch (e) {
+    /* We'll return original source */
```
Comment (KhadijaMahanga) [inline src/utils/fetchOpenAfricaDatasets.js:68]: continue?

**Review by KhadijaMahanga — APPROVED** [review]
Looks good @kilemensi

File: src/utils/fetchOpenAfricaDatasets.js (hunk 2 above)
Comment (kilemensi) [reply src/utils/fetchOpenAfricaDatasets.js:68]: Isn't `continue only useful` in loops?

**Pull request merged**